ddtrace/contrib/tornado/decorators.py

```diff
@@ -25,7 +26,14 @@ def _finish_span(future):
             # retrieve the exception from the Future object
             # that is executed in a different Thread
             if future.exception():
-                span.set_exc_info(*sys.exc_info())
+                exc = future.exception()
+                exc_type = type(exc)
+                exc_type_str = '%s.%s' % (exc_type.__module__, exc_type.__name__)
+                exc_tb = '%s: %s' % (exc_type.__name__, str(exc))
```

Comment:
Did you look at the numerous functions the `traceback` module provides? It might do this for you in a safer fashion.

Maybe https://docs.python.org/3/library/traceback.html#traceback.format_exc